### PR TITLE
[Project] Fail sync functions silently on workflow runs

### PR DESCRIPTION
**Pull request by alonmr**
Sync functions reloads functions that are not necessarily related to the workflow, therefore it may fail on issues like file not found for functions that are not needed. This changes sync functions to fail silently when there is a failure to reload a function when trying to run a workflow. If a function fails to load and is later needed in the workflow, it shall be fetched from the DB or fail if it doesn't exist.

https://iguazio.atlassian.net/browse/ML-7706

### CODE REVIEW

File: mlrun/projects/project.py [hunk 1]
```diff
@@ -2810,47 +2810,81 @@ def push(
...
+                        message += f" Skipping {name} function reload."
+                        logger.warn(message)
+                        continue
+
+                    raise mlrun.errors.MLRunMissingDependencyError(message) from exc
+            else:
+                message = f"Function {name} must be an object or dict."
+                if silent:
+                    message += f" Skipping {name} function reload."
+                    logger.warn(message)
```
Comment (liranbg) [inline mlrun/projects/project.py]: why not abstract the log and move parameters to kwarg? is it possible? I know this warning is for clients but still we need to be consistent with log structure as much as possible

File: mlrun/projects/project.py [hunk 2]
```diff
@@ -2810,47 +2810,81 @@ def push(
...
-        """reload function objects from specs and files"""
+    def sync_functions(
+        self,
+        names: list = None,
+        always: bool = True,
+        save: bool = False,
+        silent: bool = False,
+    ):
+        """
+        Reload function objects from specs and files.
```
Comment (liranbg) [inline mlrun/projects/project.py:2821]: worth adding _what_ does it sync and against what (and precedence, etc)

File: server/api/db/sqldb/db.py [hunk 3]
```diff
@@ -4637,7 +4637,7 @@ def _delete(self, session, cls, **kw):
             session.delete(obj)
         session.commit()
 
-    def _find_lables(self, session, cls, label_cls, labels):
+    def _find_labels(self, session, cls, label_cls, labels):
```
Comment (liranbg) [inline server/api/db/sqldb/db.py:4640]: 👏🏼

**Review by liranbg — CHANGES_REQUESTED** [review]
minor

File: mlrun/projects/project.py [hunk 4]
```diff
@@ -2810,47 +2810,92 @@ def push(
...
-                raise ValueError(f"function named {name} not found")
+            function_definition = self.spec._function_definitions.get(name)
+            if not function_definition:
+                if silent:
+                    logger.warn(
+                        "Function definition was not found, skipping reload", name=name
+                    )
+                    continue
+
+                raise ValueError(f"Function named {name} not found")
```
Comment (rokatyy) [inline mlrun/projects/project.py:2858]:
```suggestion
            if not function_definition:
                if not silent:
                    raise ValueError(f"Function named {name} not found")
                logger.warn(
                    "Function definition was not found, skipping reload", name=name
                )
```

File: mlrun/projects/project.py [hunk 5]
```diff
@@ -2810,47 +2810,92 @@ def push(
...
-                    raise mlrun.errors.MLRunMissingDependencyError(
-                        f"File {exc.filename} not found while syncing project functions"
-                    ) from exc
+                    message = f"File {exc.filename} not found while syncing project functions."
+                    if silent:
+                        message += " Skipping function reload"
+                        logger.warn(message, name=name)
+                        continue
+
+                    raise mlrun.errors.MLRunMissingDependencyError(message) from exc
```
Comment (rokatyy) [inline mlrun/projects/project.py:2884]: same

File: mlrun/projects/project.py [hunk 6]
```diff
@@ -2810,47 +2810,92 @@ def push(
...
+            else:
+                message = f"Function {name} must be an object or dict."
+                if silent:
+                    message += " Skipping function reload"
+                    logger.warn(message, name=name)
+                    continue
+                raise ValueError(message)
```
Comment (rokatyy) [inline mlrun/projects/project.py:2891]: same

**Review by rokatyy — COMMENTED** [review]
Minor

File: mlrun/projects/project.py (hunk 4 above)
Comment (alonmr) [reply mlrun/projects/project.py:2858]: actually I still need to continue because I need to move on to the next function so I kept it as is

**Review by rokatyy — APPROVED** [review]

**Pull request merged**